kinto/__main__.py

```diff
@@ -52,13 +52,21 @@ def main(args=None):
                               default=False)
     parser_start.set_defaults(which='start')
 
-    args = vars(parser.parse_args())
+    try:
+        args = vars(parser.parse_args(args))
+    except SystemExit:  # pragma: no cover for python2
+        return 10
+
     config_file = args['ini_file']
 
+    if 'which' not in args:  # pragma: no cover
```

Comment:
It looks like this is caused by a bug in the argparse module: http://bugs.python.org/issue9253#msg186387

The workaround is to set a `dest` and set `subparsers.required = True`. I have done this in https://github.com/Kinto/kinto/pull/576/commits/84f74af1bd32e9cda251e64a38ae51ce813ddc80.